### ISSUE TITLE
Inverse of SpacingD transform can't work with SegmentationSaver resample

### ISSUE THREAD

**Issue by Nic-Ma** [Project-MONAI/MONAI#2011]
**Describe the bug**
If we set `resample=True` in the SegmentationSaver and save the inverted results of SpacingD transform, it will output a data with wrong content, seems somewhere modified `affine`.

**Comment by Nic-Ma** [comment]
I will take a look at this issue ASAP.
Thanks.

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Actually, I checked the code and I think there is no issue with the resampling logic in SegmentationSaver.
We should set `resample=False` when saving inverted data, if set to True, it will fetch the `affine` of `image_meta_dict` from `engine.state.batch`, **not the inverted meta_dict of `pred`**. So `affine` is not equal to `original_affine`.
I submitted PR to give user a warning message for this case: https://github.com/Project-MONAI/MONAI/pull/2013

If we want to use the inverted meta_dict, we should change the `batch_transform` logic of `SegmentationSaver` to fetch meta_dict from `engine.state.output` instead of `engine.state.batch`. This will be within the topic for v0.6 workflows to combine `engine.state.batch` and `engine.state.output`.

Thanks.
